test/integration-tests/strategies/delta-strategy-short-put.ts

```diff
@@ -0,0 +1,412 @@
+    });
+
+    it('should revert when trying to reduce a safe position', async () => {
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry);
+      expect(fullCloseAmount).to.be.eq(0);
+      await expect(vault.connect(randomUser).reducePosition(positionId, toBN('10000'))).to.be.revertedWith(
+        'amount exceeds allowed close amount',
+      );
+    });
+
+    it('reduce full position if unsafe position + delta is in range', async () => {
+      // 13% crash
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('2600'), 'sETH');
+      const positionId = await strategy.strikeToPositionId(strikes[2]); // 2700 strike
+      const preReduceBal = await susd.balanceOf(strategy.address);
+
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry.sub(10)); //account for time passing
+      expect(fullCloseAmount).to.be.gt(0);
+      await vault.connect(randomUser).reducePosition(positionId, fullCloseAmount);
+      const postReduceBal = await susd.balanceOf(strategy.address);
+      expect(postReduceBal).to.be.lt(preReduceBal);
+    });
+
+    it('partially reduce position if unsafe position + delta is in range', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('2600'), 'sETH');
+      const preReduceBal = await susd.balanceOf(strategy.address);
+
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry.sub(10)); //account for time passing
+      expect(fullCloseAmount).to.be.gt(0);
+      await vault.connect(randomUser).reducePosition(positionId, fullCloseAmount.div(2));
+      const postReduceBal = await susd.balanceOf(strategy.address);
+      expect(postReduceBal).to.be.lt(preReduceBal);
+    });
+
+    it('revert reduce position if unsafe position + close amount too large', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('2250'), 'sETH');
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry.sub(10)); //account for time passing
+      expect(fullCloseAmount).to.be.gt(0);
+      await expect(vault.connect(randomUser).reducePosition(positionId, fullCloseAmount.mul(2))).to.be.revertedWith(
+        'amount exceeds allowed close amount',
+      );
+    });
+
+    it('partially reduce position with force close if delta out of range', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('2000'), 'sETH');
+
+      const [positionBefore] = await lyraTestSystem.optionToken.getOptionPositions([positionId]);
+
+      const fullCloseAmount = await strategy.getAllowedCloseAmount(position, strikePrice, expiry.sub(10)); //account for time passing
+      expect(fullCloseAmount).to.be.gt(0);
+
+      // send strategy some usdc so they can successfully reduce position
+      await susd.mint(strategy.address, toBN('50000'));
```

Comment:
would rather reduce position by a smaller amount since minting funds to strategy isn't very realistic.

Reply:
the reason i used it is because using a small amount will leave the vault still liquidatable after reducePosition, so the call reverts with `Minimum collateral not met`.

Reply:
I imagine in this scenario, we will need to send money to the strategy and then call reducePosition.

Reply:
got it, that makes sense. Forgot Avalon doesn't let u reduce position if the final state isn't > minCollat. Since we'll usually have plenty time to reduce position if it's risky, probably fine to leave as is then.